[WIP] Updated to CSI v1.0.0

mcronce opened the pull request:
I don't consider this ready to be merged yet, having not yet tested the RBD plugin.  I'm submitting it now, primarily to request feedback.  The cephfs plugin _is_ tested and working.

rootfs commented:
thanks @mcronce. Have you tried it on Kubernetes 1.12 too? If 1.12 is not working, we can create a branch for 1.13 and still continue rest of the work in master branch. 

cc @mkimuram @mickymiek for potential merge conflicts

rootfs commented:
there is a gofmt and go vet error
```
0.03s$ test -z $(gofmt -s -l $GO_FILES)
The command "test -z $(gofmt -s -l $GO_FILES)" exited with 1.
0.53s$ go vet -v $(go list ./... | grep -v /vendor/)
cephfs/main.go:24:2: could not import github.com/ceph/ceph-csi/pkg/cephfs (can't find import: "github.com/ceph/ceph-csi/pkg/cephfs")
```

mcronce commented:
Re: Kubernetes 1.12, almost definitely not.  I know with certainty that driver registration fails on 1.12 - that's the reason I waited until today to submit this, I didn't want to go building kubelets from source and dropping them in my cluster, and I didn't want to submit a PR before at least _slightly_ testing :)

I'm not sure if the plugin will fail to act, though.  My _guess_ would be no, since there are protobuf changes between CSI 0.3.0 and CSI 1.0.0.

As far as the gofmt/go vet errors, I'll get them fixed up

mcronce commented:
Okey-dokey, so, this round of changes includes the following:
- Ran `gofmt` on `pkg/rbd/controllerserver.go`
- Fixed up errors in `pkg/rbd/rbd.go` that `go vet` mentioned
- Changed the cephfs provisioner to utilize the `name` attribute provided in the `CreateVolumeRequest` (admittedly, this was the primary reason I pulled down the project and started work on it for my own needs haha) - CSI 1.0.0 mentions this at https://github.com/container-storage-interface/spec/blob/v1.0.0/csi.proto#L176 - I will push another commit that makes the same changes to the RBD plugin for volumes and snapshots once I get setup to test the RBD plugin
  - CSI 1.0.0 is less explicit about it for snapshots, but my interpretation of the note at https://github.com/container-storage-interface/spec/blob/v1.0.0/csi.proto#L843 is that it should be the same as volume names.  If anybody disagrees I'll leave that part out :)

I didn't get the same result from `go vet` that you did - after fixing what I got in `pkg/rbd/rbd.go`, I've got empty output:

```
$ go list ./... | grep -v '^/vendor'
github.com/ceph/ceph-csi/cephfs
github.com/ceph/ceph-csi/pkg/cephfs
github.com/ceph/ceph-csi/pkg/rbd
github.com/ceph/ceph-csi/rbd
$ go vet $(go list ./... | grep -v '^/vendor')
$
```

I do have the project checked out to `${GOPATH}/src/github.com/ceph/ceph-csi`, would that make a difference?

gman0 commented on Makefile:
```diff
@@ -15,10 +15,10 @@
 
 CEPHFS_IMAGE_NAME=$(if $(ENV_CEPHFS_IMAGE_NAME),$(ENV_CEPHFS_IMAGE_NAME),quay.io/cephcsi/cephfsplugin)
-CEPHFS_IMAGE_VERSION=$(if $(ENV_CEPHFS_IMAGE_VERSION),$(ENV_CEPHFS_IMAGE_VERSION),v0.3.0)
+CEPHFS_IMAGE_VERSION=$(if $(ENV_CEPHFS_IMAGE_VERSION),$(ENV_CEPHFS_IMAGE_VERSION),v1.0.0)
```
If we could refrain from tagging the cephfs plugin as 1.0 just yet and instead have this as let's say 0.9. I wanted to use this major version bump as an opportunity to make some compatibility-breaking changes (e.g. getting rid of the `provisionVolune` attribute), but @mcronce was faster with this PR than me ;) 

@rootfs ?

rootfs replied:
@mcronce @gman0 what if we get a csi-v1.0 branch and commit the changes there?

gman0 replied:
Sounds good!

rootfs replied:
@mcronce @gman0 i created csi-v1.0 branch https://github.com/ceph/ceph-csi/tree/csi-v1.0

rootfs replied:
@mcronce can you open an PR against csi-v1.0 branch? We'll change the Makefile as well to create container images using csi-v1.0 tag

mcronce replied:
@rootfs Done - #105

mcronce commented:
Closing in favor of #105
